timemachine/cpp/src/bd_exchange_move.cu

```diff
@@ -115,30 +124,41 @@ void BDExchangeMove<RealType>::move(
             cudaMemcpyDeviceToDevice,
             stream));
 
-        // The d_translation_ buffer is [x,y,z,w] where [x,y,z] are a random translation and w is used for acceptance
-        curandErrchk(templateCurandUniform(cr_rng_, d_translations_.data, d_translations_.length));
-
         // We only ever sample a single molecule
-        sampler_.sample_device(num_target_mols_, 1, d_log_weights_before_.data, d_samples_.data, stream);
+        sampler_.sample_given_noise_device(
+            num_target_mols_,
+            1,
+            d_log_weights_before_.data,
+            d_sample_noise_.data + (move * num_target_mols_),
+            d_sampling_intermediate_.data,
+            d_samples_.data,
+            stream);
 
         // Don't move translations into computation of the incremental, as different translations can be used
         // by different bias deletion movers (such as targeted insertion)
         // scale the translations as they are between [0, 1]
-        this->compute_incremental_weights(N, true, d_coords, d_box, this->d_quaternions_.data + noise_offset_, stream);
+        this->compute_incremental_weights(
+            N,
+            true,
+            d_box,
+            d_coords,
+            this->d_quaternions_.data + (move * QUATERNIONS_PER_STEP),
+            this->d_translations_.data + (move * BD_TRANSLATIONS_PER_STEP_XYZW),
+            stream);
 
         logsumexp_.sum_device(num_target_mols_, d_log_weights_after_.data, d_log_sum_exp_after_.data, stream);
 
         k_attempt_exchange_move<RealType><<<ceil_divide(N_, tpb), tpb, 0, stream>>>(
             N,
-            d_translations_.data + 3, // Offset to get the last value for the acceptance criteria
+            d_translations_.data + (move * BD_TRANSLATIONS_PER_STEP_XYZW) +
+                3, // Offset to get the last value for the acceptance criteria
```

Comment:
nit: would it make sense to define an enum similar to https://github.com/proteneer/timemachine/blob/c0a4d30eee8d557f14024bef83952eee23624072/timemachine/cpp/src/nonbonded_common.hpp#L12 to avoid a "magic number" here?

Reply:
Removed the magic numbers in https://github.com/proteneer/timemachine/pull/1234/commits/10d157b6f3c366f3e80a6559ca229ed3339821eb just subtracted 1 from the values rather than an enum since none of the other definitions in the enum would be used